### PR TITLE
CI fix doc deploy (from: #1071) 

### PR DESCRIPTION
**Pull request by adrinjalali**
closes #1071 
closes https://github.com/fairlearn/fairlearn/pull/1067

### CODE REVIEW

**Comment by adrinjalali** [comment]
This is ready: @romanlutz @riedgar-ms

File: .circleci/config.yml [hunk 1]
```diff
@@ -93,69 +93,22 @@ jobs:
...
-            chmod 700 .ssh
-            cp ssh_known_hosts .ssh/known_hosts
-            chmod 600 .ssh/known_hosts
-
-      - run:
-          name: 'Push to GitHub Pages repository'
-          # Next is only acceptable because this is running in a docker container which is not persisted
-          # See https://github.com/tschaub/gh-pages/issues/422
-          command: gh-pages --dotfiles --message "Updates" --dist docs/_build/html --repo https://github.com/fairlearn/fairlearn.github.io.git --branch master
+            if [[ "${CIRCLE_BRANCH}" =~ ^main$|^[0-9]+\.[0-9]+\.X$|adrin-testing ]]; then
```
Comment (romanlutz) [inline .circleci/config.yml:109]: `adrin-testing` is that meant to be replaced?

File: .circleci/config.yml [hunk 2]
```diff
@@ -168,6 +121,6 @@ workflows:
       - deploy-doc:
           requires:
             - build-doc
-          filters:
-            branches:
-              only: main  # don't deploy to GitHub pages from other branches
+          #filters:
```
Comment (romanlutz) [inline .circleci/config.yml:124]: This worries me. The filter seems necessary

File: devops/circle/push_doc.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+MSG="Pushing the docs to . for branch: $CIRCLE_BRANCH, commit $CIRCLE_SHA1"
+
+cd $HOME
+if [ ! -d $DOC_REPO ];
+then git clone --depth 1 --no-checkout "git@github.com:fairlearn/"$DOC_REPO".git";
+fi
+cd $DOC_REPO
+
+cp -R "$GENERATED_DOC_DIR"/* .
+git config user.email "adrin.jalali@gmail.com"
```
Comment (romanlutz) [inline devops/circle/push_doc.sh:35]: Do you want your personal email address here? I don't care, just asking.

File: .circleci/config.yml [hunk 4]
```diff
@@ -93,69 +93,22 @@ jobs:
...
-          # ssh-keygen -t rsa -m pem -C "ci-build@fairlearn.org"
-          # Do not add a passphrase.
-          # Upload public key as a deploy key to the website repository.
-          # Upload private key to CircleCI as an SSH key at
-          # https://app.circleci.com/settings/project/github/fairlearn/fairlearn#ssh.
-          fingerprints:  # fingerprint of deploy (public) key
-            - "b5:72:60:33:2a:d4:bb:28:0d:36:23:86:74:d3:c6:10"
-
+          fingerprints:
+            - "40:36:a9:4d:ea:9d:d9:94:60:0f:11:b5:fa:6a:b1:5d"
```
Comment (riedgar-ms) [inline .circleci/config.yml:101]: New key?

**Review by riedgar-ms — APPROVED** [review]
If this works, I don't have any particular objections

File: .circleci/config.yml (hunk 4 above)
Comment (adrinjalali) [reply .circleci/config.yml:101]: I wanted to make sure I have a control on everything to test things, can remove it and revert to the old key probably. I'll test.

File: .circleci/config.yml (hunk 1 above)
Comment (adrinjalali) [reply .circleci/config.yml:109]: I'll remove it, only to test the workflow.

File: .circleci/config.yml (hunk 2 above)
Comment (adrinjalali) [reply .circleci/config.yml:124]: I'll put it back, but the script skips the deploy if it's a wrong branch anyway.

File: devops/circle/push_doc.sh (hunk 3 above)
Comment (adrinjalali) [reply devops/circle/push_doc.sh:35]: it's just to match the key, can revert to the old email address once I make sure the key works.

File: .circleci/config.yml [hunk 5]
```diff
@@ -93,69 +93,22 @@ jobs:
...
     docker:
-      - image: cimg/node:14.0.0  # npm required for gh-pages
-
+      - image: cimg/python:3.8.12
     steps:
       - checkout
- 
       - add_ssh_keys:
-          # To generate a key run the following in git bash (on Windows):
-          # ssh-keygen -t rsa -m pem -C "ci-build@fairlearn.org"
```
Comment (adrinjalali) [inline .circleci/config.yml:103]: this is not very secure though, better create a ed25519 key. It's supported, even though implied otherwise.

**Comment by adrinjalali** [comment]
This should be ready to merge for whoever wants the honers :)

**Pull request merged**